zarr/core.py

```diff
@@ -341,7 +340,14 @@ def _flush_metadata_nosync(self):
             filters=filters_config,
         )
         if getattr(self._store, "_store_version", 2) == 2:
-            meta.update(dict(chunks=self._chunks, dtype=self._dtype, order=self._order))
+            meta.update(
```

Comment:
Thanks, @ziw-liu! Moving forward, I do wonder if we don't want to encapsulate all of these items to prevent this type of error.